server.ts

```diff
@@ -26,11 +26,223 @@ app.use(cors()) //add CORS support to each following route handler
+    const resourceData = req.body
+    const col_names = 'submitter,title,author,url,summary,recommendation_option,recommendation_text'
+    const values = [resourceData.submitter, resourceData.title, resourceData.author, resourceData.URL, resourceData.summary, resourceData.reccomendationOptions, resourceData.reccomendationText]
+  
+    const checkIfResourceExists = await client.query('SELECT * FROM resources WHERE url = $1',[resourceData.URL])
+  
+    if (checkIfResourceExists.rows.length!==0) {
+      res.status(250).json('already submitted before')
+    }
+    else{
+      const insertResponse =  await client.query(`INSERT into resources(${col_names}) VALUES($1,$2,$3,$4,$5,$6,$7) RETURNING *`,values)
+      let resp  = {};
+      insertResponse.rows.map(item => {
+      resp = {
+        resourceID: item.resource_id,
+        submitter: item.submitter,
+        title: item.title,
+        author: item.author,
+        URL: item.url,
+        timestamp: item.time_stamp,
+        summary: item.summary,
+        reccomendationText: item.reccomendation_text,
+        reccomendationOptions: item.reccomendation_options,
+      }
+    })
+    res.status(200).json(resp)
+  }
+  } 
+  catch(error) {
+    res.status(500).send('error')
+    console.error(error)
+  }
+});
+
+app.post("/tablename/:name", async(req, res)=>{
+  try {    
+    const acceptedTableNames = ['buildweek_resource','content_types_resource','tag_resource']    
+    const tablesData = req.body
+    const tableName = req.params.name   
+    if (tableName === "buildweek_resource"){
+      console.log(tablesData)
+      const insertResponse = await client.query(`INSERT INTO ${tableName} VALUES ($1, $2)`, [tablesData.build_week_name, tablesData.resource_id]);
+      res.status(200).json(insertResponse.rows);
+    }
+    else if (tableName === "content_types_resource"){
+      const insertResponse = await client.query(`INSERT INTO ${tableName} VALUES ($1, $2)`, [tablesData.content_type, tablesData.resource_id]);
+      res.status(200).json(insertResponse.rows);
+    }
+    else if (tableName === "tag_resource"){
+      const tagNames = tablesData.tag_name
+      for (let tag of tagNames) {
+        const insertResponse = await client.query(`INSERT INTO ${tableName} VALUES ($1, $2)`, [tag, tablesData.resource_id]);
+      } 
+      res.status(200).json("Success");
+    }
+    else {
+        res.status(500).send('table does not exist')
+    }
+  } catch(error) {
+    res.status(500).send('error')
+    console.error(error)
+  }
+});
+
+app.post("/resources/comments/:resourceid", async (req,res)=>{
+  try{
+    const {resourceid}= req.params;
+    const {user_id, message}= req.body
+    const text = 'INSERT into comment_inputs(user_id, message, resource_id) VALUES($1, $2, $3) returning *'
+    const values= [user_id, message, resourceid]
+    const postComment= await client.query(text, values)
+    res.json(postComment.rows)
+  }
+  catch(error) {
+    res.status(500).send('error')
```

Comment:
Generally would be better to have a clearer error message - makes it easier to debug in the future!